### ISSUE TITLE
How to enable javax.persistence.schema-generation.database.action create

### ISSUE THREAD

**Issue by mores** [rhuan080/multitenancyJpaJakartaEE#3]
Is there a way to modify this code to eliminate the need to manually create all the databases ?

How can Hibernate forward engineering be used to create testdb, testdb2, testdb3 ?

**Comment by rhuan080** [comment]
Hi @mores, 

You can used any hibernate/JPA properties. With this, you can configure the javax.persistence.schema-generation.database.action. Have you tested it? Did you have a issue configuring the javax.persistence.schema-generation.database.action=create?

**Comment by mores** [comment]
It only seemed to work on 1 of the databases - not all of them.

**Comment by rhuan080** [comment]
Hi,

I think this behavior is happening because the persistence unit is read just one time as the EntityManagerFactory is being injected via CDI.

` @PersistenceUnit
    protected EntityManagerFactory emf;`

Getting the EntityManagerFactory via CDI programmatically should solve the problem. However, it has a tradeoff, as each operation will run the create script on the database.

**Comment by mores** [comment]
Do you know if there is any other way to trigger the hibernate create ?

**Comment by rhuan080** [comment]
No. You will need to create one EntityManagerFactory per database. One way to do that and avoid the tradeoff is creating a map of EntityManagerFactory per database.

**Comment by mores** [comment]
It seems I can accomplish the same thing using a much simpler solution, can you identify a flaw with it ?

1. Define all the databases in persistence.xml with different name
    `<persistence-unit name="AmericaDB">`
    `<persistence-unit name="EuropaDB">`
2. Inject all databases as EntityManagers
     `@PersistenceContext(unitName = "AmericaDB")`
     `private EntityManager america;`

     `@PersistenceContext(unitName = "EuropaDB")`
     `private EntityManager europa;`    

3. When the tenant is passed in - return the correct EntityManager.

No need to call any hibernate method. And all databases will have the tables created automatically.

**Comment by rhuan080** [comment]
Hi @mores , I`m sorry for my delay! :-)

Yes, if you add one persistence-unit per DB it will work, however, in some scenarios it is not so good, because when you add a new database you will need to update the persistence.xml. For example. Imagine you want to create one database per user. In case of a new insert of the user, you`ll need to update the persistence.xml and the code. Thus, the multitenant is not scalable.

**Comment by mores** [comment]
For the same scenario - in your code base you would need to update DatabaseTenantResolver.java
then manually create the schema - I would say that is not scalable either.

I keep thinking there has to be a better solution.

**Comment by rhuan080** [comment]
Hi,
In my example, I created the Map of user and database, but I didn't provide a `setDatabase`, but it is possible. Thus, in the case of a new insert, a new database is put to map. I didn't do that because it was not the goal in my article, but you can do that. 

```
public DatabaseTenantResolver(){
        userDatasourceMap = new HashMap();
        userDatasourceMap.put("default", "MyDataSource");
        userDatasourceMap.put("america", "AmericaDB");
        userDatasourceMap.put("europa", "EuropaDB");
        userDatasourceMap.put("asia", "AsiaDB");
   }
```
